server/src/inputs/plugins/universe/planets.ts

```diff
@@ -0,0 +1,202 @@
+        
+        //I haven't yet added this to the pubsub list
+        pubsub.publish("pluginStar", {
+            pluginId: params.pluginId,
+            starId: star.name,
+        });
+
+        return planet;
+    },
+    pluginPlanetDelete(
+        context: DataContext,
+        params: {
+            pluginId: string;
+            starId: string;
+            planetId: string;
+        }
+    ) {
+        const star = getStar(
+            context,
+            params.pluginId,
+            params.starId
+        );
+        const planet = star.planets.find(s => s.name === params.planetId);
+        if (!star) {
+            throw new Error(`No planet found with id ${params.planetId}`);
+        }
+        star.planets = star.planets.filter(s => s !== planet);
+        pubsub.publish("pluginStar", {
+            pluginId: params.pluginId,
+            starId: star.name,
+        });
+        return planet;
+    },
+    pluginPlanetUpdate(
+        context: DataContext,
+        params: {
+            pluginId: string;
+            starId: string;
+            planetId: string;
+            name?: string;
+            age?: number,
+            classification?: PlanetTypes,
+            radius?: Kilometer,
+            terranMass?: TerranMass,
+            isHabitable?: boolean,
+            lifeforms?: string[],
+            zone?: Zone,
+            atmosphericComposition?: AtmosphericComposition,
+            hasClouds?: number,
+            hasRings?: number,
+            population?: number,
+            temperature?: Kelvin,
+        }
+    ) {
+        const star = getStar(
+            context,
+            params.pluginId,
+            params.starId
+        );
+        const planet = star.planets.find(s => s.name === params.planetId);
+        if (!planet) {
+            throw new Error(`No planet found with id ${params.planetId}`);
+        }
+        if (params.name) {
+            planet.name = params.name;
+        }
+        if (params.age) {
+            planet.isPlanet.age = params.age;
+        }
+        //I wasn't sure if all of these should be editable since the planet's class influences most of these attributes.
+        if (params.classification) {
+            planet.isPlanet.classification = params.classification;
+        }
+        if (params.radius) {
+            planet.isPlanet.radius = params.radius;
+        }
+        if (params.terranMass) {
+            planet.isPlanet.terranMass = params.terranMass;
+        }
+        if (params.isHabitable) {
+            planet.isPlanet.isHabitable = params.isHabitable;
+        }
+        if (params.lifeforms) {
+            planet.isPlanet.lifeforms = params.lifeforms;
+        }
+        if (params.zone) {
+            planet.isPlanet.zone = params.zone;
+        }
+        if (params.atmosphericComposition) {
+            planet.isPlanet.atmosphericComposition = params.atmosphericComposition;
+        }
+        if (params.hasClouds) {
```

Comment:
With any of these that could be the number 0, if the user passes 0 as the updated value, the if statements won't run. You should check to see if the property exists using the `in` operator, or you can check if the `typeof` is `number`.